substrate/frame/assets/src/traits.rs

```diff
@@ -0,0 +1,26 @@
+pub mod sufficients {
```

Comment:
`sufficients` isn't really a good name because the mod has nothing to do with the behavior of sufficient assets, but merely setting the `sufficient` property to `true` or `false`.

I would therefore call it `sufficiency`.

I also question whether this should be in the Assets pallet at all and not frame-support like the [Account Touch](https://github.com/paritytech/polkadot-sdk/blob/0b74812/substrate/frame/support/src/traits/misc.rs#L1181) trait.

Reply:
@muharem and @liamaharon have done some recent work on this pallet, maybe they can comment.

Reply:
I get it. Just resolved all the other changes.

As for whether `sufficiency` should be on Assets pallet and not on `frame-support`, my hint was this @muharem's [comment](https://github.com/paritytech/polkadot-sdk/pull/1768#discussion_r1366792128):

> not sure if this concept is general enough, I cannot really say.

This is because this concept might not be general enough, but is somewhat specific to Assets. Instead, the most general concept, actually, is the [`AccountTouch`](https://github.com/paritytech/polkadot-sdk/blob/0b74812/substrate/frame/support/src/traits/misc.rs#L1181) trait.

> but an alternative can be a separate trait on Assets pallet level, which lets you inspect some additional characteristics of an asset.

That's what I based on to close #1768 and open this PR instead. 😁